buildSrc/src/main/java/org/elasticsearch/gradle/transform/UnpackTransform.java

```diff
@@ -41,6 +42,20 @@
         @Optional
         List<String> getKeepStructureFor();
 
+        /**
+         * Mark output as handled like a filetree meaning that
+         * each file will be part of the output and not the singular ouptut directory.
+         * */
+        @Input
+        boolean getAsFiletreeOutput();
```

Comment:
Why would we prefer one or the other to be the case?

Reply:
This allows us at https://github.com/elastic/elasticsearch/blob/4abfb66a85d27df16edfdffcc0708e0255ab51c0/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java#L1336 to filter through all the files in the extracted module. otherwise we would only see the root dir here and keep this file collection lazy evaluated